pkg/sql/catalog/replication/reader_catalog_test.go

```diff
@@ -64,21 +66,27 @@ func TestReaderCatalog(t *testing.T) {
 		"CREATE TABLE t2(n int);",
 	}
 	for _, ddl := range ddlToExec {
-		_, err = srcConn.Exec(ddl)
-		require.NoError(t, err)
+		srcRunner.Exec(t, ddl)
 	}
 
 	now := ts.Clock().Now()
 	idb := destTenant.InternalDB().(*sql.InternalDB)
 	require.NoError(t, replication.SetupOrAdvanceStandbyReaderCatalog(ctx, serverutils.TestTenantID(), now, idb, destTenant.ClusterSettings()))
 
-	srcRunner := sqlutils.MakeSQLRunner(srcConn)
 	destRunner := sqlutils.MakeSQLRunner(destConn)
 
-	compareConn := func(query string) {
+	compareConnAt := func(query string, now hlc.Timestamp, isEqual bool) {
```

Comment:
nit: i think it would be clearer if the underlying func never accepted a timestamp and always used the captured `now` var. `compareConnAt` could instead be `check` and `compareConn` could be `checkEqual` or something. 

Now before any cluster of "checks" you always set `now` to the desired timestamp.